lib/puppet-strings/yard/parsers/puppet/statement.rb

```diff
@@ -134,13 +134,15 @@ def initialize(object, file)
   # Implements the Puppet function statement.
   class FunctionStatement < ParameterizedStatement
     attr_reader :name
+    attr_reader :type
 
     # Initializes the Puppet function statement.
     # @param [Puppet::Pops::Model::FunctionDefinition] object The model object for the function statement.
     # @param [String] file The file containing the statement.
     def initialize(object, file)
       super(object, file)
       @name = object.name
+      @type = object.return_type.value.capitalize if object.return_type
```

Comment:
@hlindberg added a check to make sure the method exists. Also updated to use `Puppet::Pops::Adapters::SourcePosAdapter.adapt` to extract the type.